# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/remoteworkers/RemoteWorkerTypes.ipdlh
# Commit: 9e22fbf8ab8f
# Full Hash: 9e22fbf8ab8f828b6e500a6e4994accc42c5e819
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-04-09 04:12:49
# Regressor Bug: 1899503
# File Overlap Count: 2
# Description:
#   Bug 1899503 - P1 New top-level IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger to support nsIWorkerDebugger interface for Workers in the parent process. r=asuth
#   
#   New IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger are introduced to support remote debugger of Workers on the parent thread.
#   
#   PRemoteWorkerDebuggerManager is IPC between the parent process main thread and the content process WorkerLauncher thread.
# ==============================================================================

diff -r a5880ebdc3d7 -r 9e22fbf8ab8f dom/workers/remoteworkers/RemoteWorkerTypes.ipdlh
--- a/dom/workers/remoteworkers/RemoteWorkerTypes.ipdlh	Tue Apr 08 09:20:53 2025 +0000
+++ b/dom/workers/remoteworkers/RemoteWorkerTypes.ipdlh	Tue Apr 08 09:43:33 2025 +0000
@@ -21,6 +21,8 @@
 using mozilla::OriginTrials from "mozilla/OriginTrialsIPCUtils.h";
 using mozilla::RFPTargetSet from "nsRFPService.h";
 
+[RefCounted] using class nsIPrincipal from "nsIPrincipal.h";
+
 namespace mozilla {
 namespace dom {
 
@@ -136,5 +138,26 @@
   void_t;
 };
 
+
+// Followings are data structures used to for RemoteWorkerDebugger
+
+struct RemoteWorkerDebuggerInfo {
+  bool isChrome;
+  uint32_t type;
+  nsString url;
+  uint64_t windowID;
+  nsIPrincipal principal;
+  uint64_t serviceWorkerID;
+  nsString Id;
+  nsString name;
+  nsString parentId;
+};
+
+struct RemoteWorkerDebuggerErrorInfo {
+  nsCString fileName;
+  uint32_t lineNo;
+  nsString message;
+};
+
 } // namespace dom
 } // namespace mozilla